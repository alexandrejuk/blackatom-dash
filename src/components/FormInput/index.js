import React from 'react'
import { Input } from 'antd'
import PropTypes from 'prop-types'

const FormInput = ({ label, type, value, onChange }) => ( 
 <div>
    <label>{label}</label>
    <Input type={type} value={value} onChange={onChange}/>
 </div>
)
 
FormInput.propTypes = {
  label: PropTypes.string.isRequired,
  type: PropTypes.string.isRequired,
  value: PropTypes.string.isRequired,
  onChange: PropTypes.func.isRequired,
}

export default FormInput;