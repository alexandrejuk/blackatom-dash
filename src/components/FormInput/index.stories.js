import React from 'react';

import { storiesOf } from '@storybook/react';
import { action } from '@storybook/addon-actions';
import FormInput from '.'

storiesOf('New Form', module)
  .add('new', () => <FormInput label="Name" type="text" />)
