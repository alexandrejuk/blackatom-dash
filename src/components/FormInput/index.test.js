import React from 'react'
import FormInput from '.'
import renderer from 'react-test-renderer'
import { shallow } from 'enzyme'

let wrapper = null
beforeAll(() => {
  wrapper = shallow(
    <FormInput 
      label="Name"
      type="text"
      value="132"
      onChange={() => null}
    />
  )
})

test('FormInput should render correctly', () => {
  const component = renderer.create(
    <FormInput 
      label="Name"
      type="text"
      value="132"
      onChange={() => null}
    />
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})

test('FormInput should have a label', () => {
  const label = wrapper.find('label')
  expect(label).toHaveLength(1)
  expect(label.contains('Name')).toBeTruthy()
})

test('FormInput should have a Input', () => {
  const input = wrapper.find('Input')
  expect(input).toHaveLength(1)
  expect(input.prop('type')).toBe('text')
})