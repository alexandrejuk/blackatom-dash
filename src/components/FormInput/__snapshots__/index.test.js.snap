// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should render correctly 1`] = `
<div>
  <label>
    Name
  </label>
  <input
    className="ant-input"
    disabled={false}
    onChange={[Function]}
    onKeyDown={[Function]}
    type="text"
    value="132"
  />
</div>
`;
